### PR TITLE
Support separate WebUI user / view modes

### PR DESCRIPTION
**Pull request by spacebares**
looks like this:
https://github.com/blakeblackshear/frigate/assets/57186372/46937caa-a7a9-4661-a6d7-4dd427295787

default behavior can be set like
```
ui:
  show_advanced_options: True
```

### CODE REVIEW

**Comment by spacebares** [comment]
lint complains about `web/src/context/index.jsx
  19:6  warning  React Hook useEffect has a missing dependency: 'showAdvOptions'. Either include it or remove the dependency array` but if i add that in then when you refresh page it forgets cookie. 

maybe this whole thing is coded wrong, i haven't a clue, react is a wonder to behold...

**Comment by NickM-27** [comment]
I think this is better presented as `Admin Mode` and the configuration option can be `default_mode` with options `admin` or `user`

**Comment by spacebares** [comment]
that would imply more than simple on/off. Could be user/advanced/admin/debug etc.
To support multiple view options have to rewrite this, i have some ideas.. maybe a new component that hides its children and it does all the work in there. So if you want something controlled by the view mode you just throw it in the component as a child.. i think that might work. Will try some things.

**Comment by spacebares** [comment]
some wip on many options:
https://github.com/blakeblackshear/frigate/assets/57186372/202a2a2a-a62a-4f6f-b4e0-a0ebe4883405

File: web/src/context/index.jsx [hunk 1]
```diff
@@ -1,6 +1,30 @@
...
+  const { data: config } = useSWR('config');
+
+  useEffect(() => {
+    async function load() {
+      const configValue = ViewModeTypes[config.ui.viewmode]; //fixes a load error
+      setViewMode(viewMode || configValue);
+    }
+
+    load();
+  }, [setViewMode, config]);
```
Comment (spacebares) [inline web/src/context/index.jsx:20]: yea but when i do poot it in there, everything stops working 🤷‍♂️🤷‍♂️🤷‍♂️🤷‍♂️

**Review by NickM-27 — COMMENTED** [review]
Very cool, I think that's what Blake had in mind and it's great how easily it can be expanded to other things in the future

File: web/src/components/AppBar.jsx [hunk 2]
```diff
@@ -38,7 +38,7 @@ export default function AppBar({ title: Title, overflowRef, onOverflowClick }) {
   return (
     <div
       id="appbar"
-      className={`w-full border-b border-gray-200  dark:border-gray-700 flex items-center align-middle p-2  fixed left-0 right-0 z-10 bg-white dark:bg-gray-900 transform transition-all duration-200 ${
```
[the remark this answers is not in the document]
Comment (spacebares) [reply web/src/components/AppBar.jsx:41]: to fix tooltips in the appbar 
image

Had to also move `AppBar` farther down to get it to render on top of everything [web/src/app.tsx line 28](https://github.com/blakeblackshear/frigate/blob/fff14f7dc61382804f7a27e333d2e8fbbef6d681/web/src/app.tsx#L28C27-L28C27).

**Comment by spacebares** [comment]
menu looks like this atm:
image

but i was thinking about:
1ZDSi6ZJzG

not sure which one to pick, 🤔

**Comment by NickM-27** [comment]
I prefer the former, makes it much more obvious which is the currently selected mode and takes up less space

**Comment by johnmarksilly** [comment]
Dropdown inside a menu is not ideal though. A separate icon button for "user pref" might be better. Profile icon in header expands to the view modes.

**Comment by NickM-27** [comment]
Thank you for your pull request, we are currently rewriting the UI for the next version of frigate and this PR can't be merged. This functionality is being considered for the new UI.

**Pull request closed without merging**